File: common/src/main/java/com/pluxity/facility/dto/FacilityFloorUpdateRequest.java
package com.pluxity.facility.dto;

import com.pluxity.facility.floor.dto.FloorRequest;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import java.util.List;

public record FacilityFloorUpdateRequest(
        @Schema(description = "층 정보", requiredMode = Schema.RequiredMode.REQUIRED) @NotNull
                List<FloorRequest> floors) {}

File: common/src/main/java/com/pluxity/facility/dto/FacilityFloorUpdateRequest.kt
package com.pluxity.facility.dto

import com.pluxity.facility.floor.dto.FloorRequest
import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotNull

@JvmRecord
data class FacilityFloorUpdateRequest(
    @field:Schema(
        description = "층 정보",
        requiredMode = Schema.RequiredMode.REQUIRED
    ) @param:Schema(description = "층 정보", requiredMode = Schema.RequiredMode.REQUIRED) val floors: @NotNull MutableList<FloorRequest?>?
)
